modules/post/multi/recon/local_exploit_suggester.rb

```diff
@@ -73,33 +114,71 @@ def set_module_options(mod)
     end
   end
 
-  def is_module_wanted?(mod)
-    (
-      mod.kind_of?(Msf::Exploit::Local) &&
-      mod.has_check? &&
-      is_session_type_compat?(mod) &&
-      is_module_platform?(mod) &&
-      is_module_arch?(mod) &&
-      is_module_options_ready?(mod)
-    )
+  def set_module_target(mod)
+    session_platform = Msf::Module::Platform.find_platform(session.platform)
+    target_index = mod.targets.find_index do |target|
+      # If the target doesn't define its own compatible platforms or architectures, default to the parent (module) values.
+      target_platforms = target.platform&.platforms || mod.platform.platforms
+      target_architectures = target.arch || mod.arch
+
+      target_platforms.include?(session_platform) && target_architectures.include?(session_arch)
+    end
+    mod.datastore['Target'] = target_index if target_index
   end
 
   def setup
     return unless session
 
     print_status "Collecting local exploits for #{session.session_type}..."
 
+    setup_validation_options
+    setup_color_options
+
     # Collects exploits into an array
     @local_exploits = []
-    framework.exploits.each do |name, _obj|
+    framework.exploits.each_with_index do |(name, _obj), index|
+      print "%bld%blu[*]%clr Collecting exploit #{index + 1} / #{framework.exploits.count}\r"
       mod = framework.exploits.create name
       next unless mod
+
       set_module_options mod
-      next unless is_module_wanted? mod
-      @local_exploits << mod
+      set_module_target mod
+
+      verify_result = verify_mod(mod)
+      @local_exploits << { module: mod, result: verify_result } if verify_result[:has_check]
     end
   end
 
+  def verify_mod(mod)
+    return { has_check: false } unless mod.kind_of?(Msf::Exploit::Local) && mod.has_check?
+
+    result = {
+      has_check: true,
+      is_module_platform: (@validate_platform ? is_module_platform?(mod) : true),
+      is_module_arch: (@validate_arch ? is_module_arch?(mod) : true),
```

Comment:
not a blocker: This should probably be `is_session_arch` / `is_session_platform`